compiler/cpp/src/generate/t_go_generator.cc

```diff
@@ -1233,6 +1249,46 @@ void t_go_generator::generate_isset_helpers(ofstream& out,
   }
 }
 
+/**
+ * Generates the CountSetFields helper method for a struct
+ */
+void t_go_generator::generate_countsetfields_helper(ofstream& out,
+                                            t_struct* tstruct,
+                                            const string& tstruct_name,
+                                            bool is_result) {
+  (void)is_result;
+  const vector<t_field*>& fields = tstruct->get_members();
+  vector<t_field*>::const_iterator f_iter;
+  const string escaped_tstruct_name(escape_string(tstruct->get_name()));
+
+  out << indent() << "func (p *" << tstruct_name << ") CountSetFields" << tstruct_name
```

Comment:
What is the purpose to use the struct name as a suffix for the method? To avoid name confliction?

Reply:
Hmm, this was quite some time ago. I haven't been using thrift for a while so I can't say for sure.

Reply:
OK